tests/Functional/ContentRenderTestBase.php

```diff
@@ -321,4 +322,58 @@ protected function assertLinkIcon(NodeElement $element, string $title, string $h
     $this->assertContains($icon_type, $icon[0]->getAttribute('xlink:href'));
   }
 
+  /**
+   * Creates Document reference entity.
+   *
+   * @param string $name
+   *   Entity name. Is used as a parameter for test data.
+   * @param string $bundle
+   *   Entity bundle.
+   * @param int $status
+   *   Entity status.
+   *
+   * @return \Drupal\oe_content_entity_document_reference\Entity\DocumentReference
+   *   Document reference entity.
+   */
+  protected function createDocumentReferenceEntity(string $name, string $bundle, int $status): DocumentReference {
```

Comment:
There are 2 functions in one. I would advise to split it on 2 functions createDocumentDocumentReferenceEntity and createPublicationDocumentReferenceEntity